base58/src/lib.rs

```diff
@@ -208,21 +208,21 @@ where
     let mut leading_zeroes = true;
     // Build string in little endian with 0-58 in place of characters...
     for d256 in data {
-        let mut carry = d256 as usize;
+        let mut carry = usize::from(d256);
         if leading_zeroes && carry == 0 {
             leading_zero_count += 1;
         } else {
             leading_zeroes = false;
         }
 
         for ch in buf.slice_mut() {
-            let new_ch = *ch as usize * 256 + carry;
-            *ch = (new_ch % 58) as u8;
+            let new_ch = usize::from(*ch) * 256 + carry;
+            *ch = (new_ch % 58) as u8; // cast looses data intentionally
```

Comment:
One is doing pure math (and can use `u32` which is the morally correct type) while the other is computing an array index (which forces us to use `usize` or deal with more (fallible conversions).

Reply:
> the other is computing an array index

I don't see it. The only index accesses I can see directly convert from a different value of type `u8`.

Also IIUC `u16` should be enough (but I'm not claiming it's better than `u32` - probably doesn't matter or it may be even worse on some architectures).

Reply:
Oh, you're right, I was looking at another instance of `usize::from` (the final one in the diff, which is indeed an array index).

For `new_ch` you're right that `u32` would work, and we ought to use it for consistency with the other code. And that we could in theory use `u16`.

Reply:
The question is if we want to do it in this PR. I don't mind either way.